icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -196,6 +236,25 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         if (bestSnlp[numCodePts] == kint32max) {
             t_boundary[numBreaks] = numCodePts;
             numBreaks++;
+        } else if (isPhraseBreaking) {
+            t_boundary[numBreaks] = numCodePts;
+            numBreaks++;
+            int prevIdx = numCodePts;
+            int codeUnitIdx = 0, length = 0;
+            for (int i = prev[numCodePts]; i > 0; i = prev[i]) {
+                codeUnitIdx = prenormstr.offsetByCodePoints(0, i);
```

Comment:
FYI
Please consider a follow-up task of making sure that you have good test coverage with strings with supplementary characters, for example [supplementary Katakana](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3AKatakana%3A%5D-%5B%5Cu0000-%5CuFFFF%5D&g=&i=) characters etc. that would make a difference in this code.